src/pages/library/DataBrowser.test.ts

```diff
@@ -0,0 +1,99 @@
+import * as _ from 'lodash/fp'
+import {
+  convertColsToSettings,
+  convertSettingsToCols, extractCatalogFilters
+} from 'src/pages/library/DataBrowser'
+import { TEST_DATASET_ONE, TEST_DATASET_TWO, TEST_DATASETS } from 'src/pages/library/test-datasets'
+
+
+const settings = [{ name: 'Consortiums', key: 'consortiums', visible: true },
+  { name: 'Species', key: 'species', visible: false }]
+const cols = ['consortiums']
+const expectedFilterSectionsAndValuesForTestDatasets = [
+  {
+    header: 'Access type',
+    values: ['Granted', 'Controlled', 'External', 'Pending']
+  },
+  {
+    header: 'Consortium',
+    values: ['The Dog Land', 'Cats R Us']
+  },
+  {
+    header: 'Data use policy',
+    values: ['DUO:0000042']
+  },
+  {
+    header: 'Data modality',
+    values: ['Epigenomic', 'Genomic', 'Transcriptomic']
+  },
+  {
+    header: 'Assay category',
+    values: ['nuq-seq', 'RNA-seq']
+  },
+  {
+    header: 'File type',
+    values: ['bam', 'bai']
+  },
+  {
+    header: 'Disease',
+    values: ['too cute', 'aww']
+  },
+  {
+    header: 'Species',
+    values: ['dogs', 'cats']
+  }
+]
+const expectedFilterHeaders = _.map(expectedFilterSection => expectedFilterSection.header, expectedFilterSectionsAndValuesForTestDatasets)
+
+
+describe('DataBrowser', () => {
+  it('converts selected columns to settings', () => {
+    // Avoid copying entire list of columns into this test by checking for a subset of elements.
+    expect(convertColsToSettings(cols)).toEqual(expect.arrayContaining(settings))
+  })
+
+  it('converts settings to selected columns', () => {
+    expect(convertSettingsToCols(settings)).toMatchObject(cols)
+  })
+
+
+  it('generates proper filter sections for datasets', () => {
+    // Arrange is handled with test suite scoped constants
+    // Act
+    const extractCatalogFiltersResultHeaders = _.map(filter => filter.header, extractCatalogFilters(TEST_DATASETS))
+    // Assert
+    _.forEach(expectedHeader => expect(extractCatalogFiltersResultHeaders).toContain(expectedHeader), expectedFilterHeaders)
+  })
+
+  it('generates the correct values for each filter section', () => {
+    // Act
+    const extractCatalogFiltersResult = extractCatalogFilters(TEST_DATASETS)
+    // Assert
+    _.forEach(sectionAndValues => {
+      const filter = _.find(filter => filter.header === sectionAndValues.header, extractCatalogFiltersResult)
+      if (filter !== undefined) {
```

Comment:
It is needed because typescript complains about calling methods on potentially undefined types (without it the type is `FilterSection | undefined`. So it provides a compiletime error. This check ensures the typing is correct. I also tried to do it with a cast, but the cast didn't work.